sdk/src/test/java/io/dapr/it/DaprIntegrationTestingRunner.java

```diff
@@ -0,0 +1,137 @@
+
+    private DaprFreePorts daprFreePorts= new DaprFreePorts();
+    private Runtime rt = Runtime.getRuntime();
+    private Process proc;
+
+    private String successMessage;
+    private Class serviceClass;
+    private Boolean useAppPort;
+    private int sleepTime;
+    private String appName;
+
+    private DaprIntegrationTestingRunner(String successMessage, Class serviceClass, Boolean useAppPort, int sleepTime) {
+        this.successMessage = successMessage;
+        this.serviceClass = serviceClass;
+        this.useAppPort = useAppPort;
+        this.sleepTime = sleepTime;
+        this.generateAppName();
+    }
+
+    public static DaprIntegrationTestingRunner createDaprIntegrationTestingRunner(String successMessage, Class serviceClass, Boolean useAppPort, int sleepTime) {
+        return new DaprIntegrationTestingRunner(successMessage, serviceClass, useAppPort, sleepTime);
+    }
+
+
+    public DaprFreePorts initializeDapr() throws Exception {
+        daprFreePorts.initPorts();
+
+        String daprCommand=this.buildDaprCommand();
+        System.out.println(daprCommand);
+        proc= rt.exec(daprCommand);
+
+
+        final Runnable stuffToDo = new Thread(() -> {
+            try {
+                try (InputStream stdin = proc.getInputStream()) {
+                    try(InputStreamReader isr = new InputStreamReader(stdin)) {
+                        try (BufferedReader br = new BufferedReader(isr)){
+                            String line;
+                            while ((line = br.readLine()) != null) {
+                                System.out.println(line);
+                                if (line.contains(successMessage)) {
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                }
+            } catch (IOException ex) {
+                Assert.fail(ex.getMessage());
+            }
+        });
+
+        final ExecutorService executor = Executors.newSingleThreadExecutor();
+        final Future future = executor.submit(stuffToDo);
+        executor.shutdown(); // This does not cancel the already-scheduled task.
+        future.get(1, TimeUnit.MINUTES);
+        Thread.sleep(sleepTime);
+        return daprFreePorts;
+    }
+
+    private static final String DAPR_RUN = "dapr run --app-id %s ";
+    private static final String DAPR_COMMAND = " -- mvn exec:java -D exec.mainClass=%s -D exec.classpathScope=\"test\" -Dexec.args=\"-p %d -grpcPort %d -httpPort %d\"";
+
+    private String buildDaprCommand(){
+        StringBuilder stringBuilder= new StringBuilder(String.format(DAPR_RUN, this.appName))
+                .append(this.useAppPort ? "--app-port " + this.daprFreePorts.appPort : "")
+                .append(" --grpc-port ")
+                .append(this.daprFreePorts.grpcPort)
+                .append(" --port ")
+                .append(this.daprFreePorts.httpPort)
```

Comment:
I would not recommend to set these options. Rather, user code can read `DAPR_HTTP_PORT` and `DAPR_GRPC_PORT` which dapr cli sets.